python/PyQt6/core/auto_generated/qgsfeature.sip.in

```diff
@@ -307,13 +307,43 @@ Alternatively, in Python it is possible to directly set a field's value via the
 %MethodCode
     bool rv;
 
-    if ( a1Wrapper == Py_None )
+    if ( a1 == Py_None )
     {
       rv = sipCpp->setAttribute( a0, QVariant( QVariant::Int ) );
     }
+    else if ( PyBool_Check( a1 ) )
```

Comment:
This optimisation could also be made for the setitem methods, and for the other setAttribute overloads

Reply:
This is now done.